process/rewardTransaction/process.go

```diff
@@ -64,32 +71,37 @@ func (rtp *rewardTxProcessor) ProcessRewardTransaction(rTx *rewardTx.RewardTx) e
 	if rTx == nil {
 		return process.ErrNilRewardTransaction
 	}
+	if rTx.Value == nil {
+		return process.ErrNilValueFromRewardTransaction
+	}
+
+	rtp.mutRewardsForwarder.Lock()
+	err := rtp.rewardTxForwarder.AddIntermediateTransactions([]data.TransactionHandler{rTx})
```

Comment:
add only those transactions which are created by the current shard, the others can be simply processed.